detect_secrets/core/scan.py

```diff
@@ -326,14 +328,28 @@ def _scan_line(
     filename: str,
     line: str,
     line_number: int,
+    **kwargs: Any,
 ) -> Generator[PotentialSecret, None, None]:
     # NOTE: We don't apply filter functions here yet, because we don't have any filters
     # that operate on (filename, line, plugin) without `secret`
     try:
-        secrets = plugin.analyze_line(filename=filename, line=line, line_number=line_number)
+        analyze_line = plugin.__class__.analyze_line
```

Comment:
This could all use some comments explaining why it's necessary, considering we've gone from 1 line to 14 lines involving "Python magic internals" as you say

breaking it out into a helper function might help with the readability too

Reply:
in particular the part where we go from `plugin.analyze_line` -> `plugin.__class__.analyze_line` could use some explanation